lib/src/pokemon_tcg.dart

```diff
@@ -21,28 +22,32 @@ class PokemonTcgApi {
   static const _baseUrl = 'https://api.pokemontcg.io/v2';
   static const _setsUrl = '$_baseUrl/sets';
 
-  /// Gets a paginated list of all pokemon cards.
+  /// Gets a paginated list of all pokemon cards by default, with an optional search parameter.
   Future<List<PokemonCard>> getCards({
     int page = 0,
+    String query = '',
   }) async {
     http.Response response;
     if (page == 0) {
       response = await client.get(
-        Uri.parse('$_baseUrl/cards'),
+        Uri.parse('$_baseUrl/cards?q=$query'),
         headers: {
           'x-api-key': apiKey,
         },
       );
     } else {
       response = await client.get(
-        Uri.parse('$_baseUrl/cards?page=$page'),
+        Uri.parse('$_baseUrl/cards?q=$query&page=$page'),
         headers: {
           'x-api-key': apiKey,
         },
       );
     }
 
     JsonMap json = jsonDecode(response.body);
+    if (json.containsKey("error")) {
+      throw RemoteError('Invalid search parameter', json['error']);
```

Comment:
`json['error']` isn't a string which `RemoteError()` expects there, so this will fail. Instead it's a map which has a `message` key which has the value you want.

```suggestion
      throw RemoteError('Invalid search parameter', json['error']['message']);
```